netman/adapters/switches/arista.py

```diff
@@ -18,12 +20,15 @@ def __init__(self, switch_descriptor):
         super(Arista, self).__init__(switch_descriptor)
         self.switch_descriptor = switch_descriptor
 
+        url = urlparse(self.switch_descriptor.hostname)
```

Comment:
I would put this in the connect section, the switch descriptor is not promised to be immutable